doc/rst/tools/mason/mason.rst

```diff
@@ -313,9 +312,188 @@ If no query is provided, all packages in the registry will be listed.
+      platform=platform             linux, darwin, cray, bgq, etc.
+      arch=platform-os-target       shortcut for all three above
+
+    cross-compiling:
+      os=backend or os=be           build for compute node (backend)
+      os=frontend or os=fe          build for login node (frontend)
+
+    dependencies:
+      ^dependency [constraints]     specify constraints on dependencies
+
+  examples:
+      hdf5                          any hdf5 configuration
+      hdf5 @1.10.1                  hdf5 version 1.10.1
+      hdf5 @1.8:                    hdf5 1.8 or higher
+      hdf5 @1.8: %gcc               hdf5 1.8 or higher built with gcc
+      hdf5 +mpi                     hdf5 with mpi enabled
+      hdf5 ~mpi                     hdf5 with mpi disabled
+      hdf5 +mpi ^mpich              hdf5 with mpi, using mpich
+      hdf5 +mpi ^openmpi@1.7        hdf5 with mpi, using openmpi 1.7
+      boxlib dim=2                  boxlib built for 2 dimensions
+      libdwarf %intel ^libelf%gcc
+          libdwarf, built with intel compiler, linked to libelf built with gcc
+      mvapich2 %pgi fabrics=psm,mrail,sock
+          mvapich2, built with pgi compiler, with support for multiple fabrics
+
+
+So, to install ``HDF5`` version 1.10.1 built with gcc including a high level interface and without
+MPI, the following spec expression would be used::
+
+  mason external install hdf5@1.10.1%gcc+hl~mpi
+
+After the download, to ensure the package was downloaded, use ``mason external find`` which
+will list all installed external packages.
+
+The next step in creating the library is to add ``HDF5`` to our ``Mason.toml`` to inform Mason that
+we want the library included when our project is compiled. Use ``mason add --external hdf5@1.10.1%gcc``
+to add the external dependency to your ``Mason.toml`` without ever having to open up your editor.
+
+.. note:: Currently the name of the package, version, and compiler must all be included
+          in the value of the ``Mason.toml`` for external dependencies only. See ``Mason.toml``
+          below for details.
+
+``Mason.toml``
+
+.. code-block:: text
+
+   [brick]
+   name = "masonHDF5"
+   version = "0.1.0"
+   chplVersion = "1.18.0"
+   # linker flags for the library
+   compopts = "-lhdf5 -lhdf5_hl"
+
+   # must give name, version, compiler
+   [external]
+   hdf5 = "hdf5@1.10.1%clang"
+
+Now lets run our example. To check the name of all the examples available in your project use
+``mason run --example``. The output will represent whatever ``.chpl`` files are located in your
+``example/`` folder as follows::
+
+  $ mason run --example
+  --- available examples ---
+  --- hdf5Example.chpl
+  --------------------------
+
+Calling ``mason run --build --example hdf5Example.chpl`` will update the project's ``Mason.lock``
+and build and run ``hdf5Example.chpl``. For reference, the ``Mason.lock`` will look as follows where
+``MASON_HOME`` will be replaced with the path to the local ``mason-registry`` installation:
```

Comment:
I wanted to show how the dependencies are gathered by Mason through Spack. I am not sure how I would capture that in a sentence. Have a suggestion?

Reply:
When constructing the lock file (see below section),  Mason will work with Spack to gather dependencies. In the case of HDF5, at the time of this writing, the lock file will gain a dependency on `zlib` since the HDF5 Spack package depends on it. Any dependency a Spack package has will be handled in this manner.